main/preload.js

```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 /** @typedef {import('electron').IpcRendererEvent} IpcRendererEvent */
-/** @typedef {import('./typings').Activity} Activity */
+/** @typedef {import('@filecoin-station/core').ActivityEvent} ActivityEvent */
```

Comment:
I wasn't sure which was cleaner. I went with the current way because there's less code involved. I also wonder how your proposed approach will scale. Should we move all npm module types to our shared type files? Currently also the type of `app-builder-lib` is sourced from npm directly, not our type files. Or is that a different situation?

I'm happy to move the `core` type to our type files, but I'm curious to understand your reasoning better.

Reply:
I see your point. TBH, I am not sure which way is cleaner either.

Adding all npm module types to our shared type files seems too cumbersome to me. 

I would move imports of types that are used in more than one place to our typings file. But it's difficult to detect when to trigger this condition, i.e. when adding a new type import, we would need to detect if it's already used elsewhere first, which is not great at all.

Maybe we can use a different rule: types shared by frontend (rendered) and backend (main) should be defined in the shared typings file. Everything else can be imported directly from dependencies. On a case-by-case basis, we can decide to move some of the type imports to the shared typings file too, but it's not required.

WDYT?

Reply:
Ok I like this rule 👍  Will consolidate